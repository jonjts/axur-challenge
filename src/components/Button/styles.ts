import styled from 'styled-components';

export const ButtonContainer = styled.button<{ color?: string }>`
    background-color: ${({ color, theme: { colors: { primary } } }) => color || primary};
`;

export const LinkContainer = styled.button`
    background-color: transparent;
    color: ${({ theme: { colors: { secondary } } }) => secondary};

    :hover {
      opacity: 0.8;
      text-decoration: underline;
      box-shadow: none;
    }

    :active {
        filter: brightness(1);
        background-color: transparent;
        box-shadow: none;
    }

    :focus {
      box-shadow: none;
    }
`;
