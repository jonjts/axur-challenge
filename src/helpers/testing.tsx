import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
