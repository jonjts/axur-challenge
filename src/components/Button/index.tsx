import React, { memo } from 'react';

import { ButtonContainer, LinkContainer } from './styles';

type Props = {
    color?: string,
    onClick?(): void,
    disabled?: boolean,
    isLink?: boolean
};

const Button: React.FC<Props> = ({
    children,
    color,
    onClick,
    disabled,
    isLink
}) => (
    <>
        {
            !isLink ?
                <ButtonContainer
                    data-testid='btn'
                    color={color}
                    onClick={!disabled ? onClick : undefined}
                    className={disabled ? 'disabled' : ''}
                >
                    {children}
                </ButtonContainer>
                :
                <LinkContainer
                    data-testid='btn-link'
                    onClick={!disabled ? onClick : undefined}
                    className={disabled ? 'disabled' : ''}
                >
                    {children}
                </LinkContainer>
        }
    </>
);

export default memo(Button) as React.FC<Props>;