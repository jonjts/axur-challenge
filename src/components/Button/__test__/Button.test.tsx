import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import Button from '../index';

describe('Button', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Button />);
  });

  test('renders correctly', () => {
    const result = renderWithProviders(<Button />);
    expect(result.getByTestId('btn')).toBeInTheDocument();
  });

  test('renders correctly when link', () => {
    const result = renderWithProviders(<Button isLink />);
    expect(result.getByTestId('btn-link')).toBeInTheDocument();
  });

  test('renders correctly with children', () => {
    const result = renderWithProviders((
      <Button>
        <b data-testid='bold-text-id'>btn</b>
      </Button>
    ));
    expect(result.getByTestId('bold-text-id')).toBeInTheDocument();
  });
});
