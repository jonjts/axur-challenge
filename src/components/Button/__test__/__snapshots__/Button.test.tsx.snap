// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match with snapshot 1`] = `
<button
  className="sc-gsTCUz dPeWPU"
  data-testid="btn"
/>
`;
